samcli/commands/test_runner/run/cli.py

```diff
@@ -0,0 +1,244 @@
+    help="""
+An override for the S3 bucket used to store your tests and results.
+
+If used, ensure that you've given the Fargate container access to your bucket through the ContainerIAMRole in your Test Runner CloudFormation Template.
+""",
+)
+@click.option(
+    "--path-in-bucket",
+    required=False,
+    type=str,
+    default=_get_unique_bucket_directory_name(),
+    help=f"""
+Specify the path within the S3 bucket where the tests, requirements, and results will be stored.
+
+By default, a top level directory named test_run_<ISO 8601 date>/ is created and used.
+
+For example, right now it would be {_get_unique_bucket_directory_name()}/
+""",
+)
+@click.option(
+    "--ecs-cluster",
+    "ecs_cluster_override",
+    required=False,
+    type=str,
+    help="An override for the ECS Cluster in which Fargate runs your tests.",
+)
+@click.option(
+    "--subnets",
+    "subnets_override",
+    required=False,
+    cls=OptionNargs,
+    type=list,
+    help="""
+Specify a space separated list of subnets to use when starting the test-running task.
+If not specified, the Test Runner will automatically use subnets associated with your account.
+
+E.g. --subnets subnet-1 subnet-2
+""",
+)
+@pass_context
+def cli(
+    ctx,
+    runner_stack_name: str,
+    runner_template_path: Optional[click.Path],
+    env_file: Optional[click.Path],
+    test_command_options: Optional[str],
+    tests_path: click.Path,
+    requirements_file_path: click.Path,
+    bucket_override: Optional[str],
+    path_in_bucket: Optional[str],
+    ecs_cluster_override: Optional[str],
+    subnets_override: Optional[List[str]],
+) -> None:
+    """
+    `sam test_runner run` command entry point
+    """
+
+    do_cli(
+        ctx,
+        runner_stack_name=runner_stack_name,
+        runner_template_path=runner_template_path,
+        env_file=env_file,
+        test_command_options=test_command_options,
+        tests_path=tests_path,
+        requirements_file_path=requirements_file_path,
+        bucket_override=bucket_override,
+        path_in_bucket=path_in_bucket,
+        ecs_cluster_override=ecs_cluster_override,
+        subnets_override=subnets_override,
+    )
+
+
+def do_cli(
+    ctx,
+    runner_stack_name: str,
+    runner_template_path: Optional[click.Path],
+    env_file: Optional[click.Path],
+    test_command_options: Optional[str],
+    tests_path: click.Path,
+    requirements_file_path: click.Path,
+    bucket_override: Optional[str],
+    path_in_bucket: Optional[str],
+    ecs_cluster_override: Optional[str],
+    subnets_override: Optional[List[str]],
+) -> None:
+    """
+    implementation of `sam test_runner run` command
+    """
+
+    from samcli.lib.test_runner.fargate_testsuite_runner import FargateTestsuiteRunner
+    from samcli.lib.utils.boto_utils import get_boto_client_provider_with_config
+    from samcli.yamlhelper import parse_yaml_file
+
+    try:
+        other_env_vars = dict(parse_yaml_file(env_file)) if env_file else {}
```

Comment:
Doesn't `parse_yaml_file` return a dictionary?

Reply:
Yep, returns an OrderedDict, I had the cast for debugging readability and forgot to remove